include/seastar/util/log.hh

```diff
@@ -316,6 +353,58 @@ public:
     }
     /// \endcond
 
+#if FMT_VERSION >= 80000
+    /// Log with error tag
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::error(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void error(std::string_view s) noexcept {
+        log(log_level::error, "{}", s);
+    }
+    /// Log with warn tag
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::warn(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void warn(std::string_view s) noexcept {
+        log(log_level::warn, "{}", s);
+    }
+    /// Log with info tag
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::info(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void info(std::string_view s) noexcept {
+        log(log_level::info, "{}", s);
+    }
+    /// Log with info tag on shard zero only:
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::info0(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void info0(std::string_view s) noexcept {
+        if (is_shard_zero()) {
+            log(log_level::info, "{}", s);
+        }
+    }
+    /// Log with debug tag
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::debug(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void debug(std::string_view s) noexcept {
+        log(log_level::debug, "{}", s);
+    }
+    /// Log with trace tag
+    ///
+    /// \param s - message to be printed
+    ///
+    /// \sa logger::trace(fmt_string<Args...> fmt, Args&&... args) noexcept
+    void trace(std::string_view s) noexcept {
```

Comment:
Let's say someone writes "mylogger.info("blah {}"); and forgets the argument. With this, they won't get a compile error.

Reply:
Ah, I see I already asked it. I think these overloads should be removed.

Reply:
But, we should have overloads on fmt::basic_runtime\<char>, so users can create runtime formats (though it's a bad idea).

Reply:
> Let's say someone writes "mylogger.info("blah {}"); and forgets the argument. With this, they won't get a compile error.

these overloads were implemented to address https://github.com/scylladb/seastar/pull/1196#discussion_r958674035 . will re-purpose them to accept `fmt::basic_runtime<char>` instead. i still don't think a good library should encourage bad practice. we have a similar discussion at https://github.com/scylladb/seastar/pull/1196#discussion_r1172484772

Reply:
i still have trouble to add overload for `fmt::basic_runtime<char>`. seems it's difficult to implement a `is_consteval` in C++17. especially, we need to differentiate by inspecting a variable. if only we can use `__builtin_constant_p` along with `std::enable_if_t<>`...